core/src/main/java/hudson/model/ParametersDefinitionProperty.java

```diff
@@ -66,15 +66,30 @@
 public class ParametersDefinitionProperty extends OptionalJobProperty<Job<?, ?>>
         implements Action {
 
-    private final List<ParameterDefinition> parameterDefinitions;
+    private List<ParameterDefinition> parameterDefinitions;
 
     @DataBoundConstructor
-    public ParametersDefinitionProperty(List<ParameterDefinition> parameterDefinitions) {
-        this.parameterDefinitions = parameterDefinitions;
+    public ParametersDefinitionProperty(@Nonnull List<ParameterDefinition> parameterDefinitions) {
+        if (parameterDefinitions != null) {
+            this.parameterDefinitions = parameterDefinitions;
+        } else {
+            this.parameterDefinitions = Collections.emptyList();
+        }
     }
 
-    public ParametersDefinitionProperty(ParameterDefinition... parameterDefinitions) {
-        this.parameterDefinitions = Arrays.asList(parameterDefinitions);
+    public ParametersDefinitionProperty(@Nonnull ParameterDefinition... parameterDefinitions) {
```

Comment:
I'm saying this should be `@Nonnull` which is what you have, ignore the CheckForNull zealot... we can argue about which `@Nonnull` or `@NonNull` but that is a different argument and this is Jenkins core which has a habit of using the wrong annotation so we'll not fight that fight here